Index: node-v4.8.3/deps/v8/src/runtime/runtime-i18n.cc
===================================================================
--- node-v4.8.3.orig/deps/v8/src/runtime/runtime-i18n.cc
+++ node-v4.8.3/deps/v8/src/runtime/runtime-i18n.cc
@@ -22,6 +22,7 @@
 #include "unicode/dtfmtsym.h"
 #include "unicode/dtptngen.h"
 #include "unicode/locid.h"
+#include "unicode/normlzr.h"
 #include "unicode/numfmt.h"
 #include "unicode/numsys.h"
 #include "unicode/rbbi.h"
@@ -32,6 +33,7 @@
 #include "unicode/ucurr.h"
 #include "unicode/uloc.h"
 #include "unicode/unum.h"
+#include "unicode/ustring.h"
 #include "unicode/uversion.h"
 
 
