Index: node-v4.9.1/tools/test.py
===================================================================
--- node-v4.9.1.orig/tools/test.py
+++ node-v4.9.1/tools/test.py
@@ -480,10 +480,16 @@ class TestCase(object):
 
   def RunCommand(self, command, env):
     full_command = self.context.processor(command)
-    output = Execute(full_command,
+    reruns = 0
+    while(reruns < 5):
+      print("FLAKY rerun: ", command)
+      reruns += 1
+      output = Execute(full_command,
                      self.context,
                      self.context.GetTimeout(self.mode),
                      env)
+      if output.exit_code == 0:
+        break;
     self.Cleanup()
     return TestOutput(self,
                       full_command,
