Index: node-v10.10.0/deps/npm/bin/npm-cli.js
===================================================================
--- node-v10.10.0.orig/deps/npm/bin/npm-cli.js
+++ node-v10.10.0/deps/npm/bin/npm-cli.js
@@ -1,4 +1,4 @@
-#!/usr/bin/env node
+#!/usr/bin/node10
 ;(function () { // wrapper in case we're in module_context mode
   // windows: running "npm blah" in this folder will invoke WSH, not node.
   /* global WScript */
Index: node-v10.10.0/deps/npm/bin/npx
===================================================================
--- node-v10.10.0.orig/deps/npm/bin/npx
+++ node-v10.10.0/deps/npm/bin/npx
@@ -1,4 +1,3 @@
-#!/bin/sh
 (set -o igncr) 2>/dev/null && set -o igncr; # cygwin encoding fix
 
 basedir=`dirname "$0"`
Index: node-v10.10.0/deps/npm/bin/npm
===================================================================
--- node-v10.10.0.orig/deps/npm/bin/npm
+++ node-v10.10.0/deps/npm/bin/npm
@@ -1,4 +1,3 @@
-#!/bin/sh
 (set -o igncr) 2>/dev/null && set -o igncr; # cygwin encoding fix
 
 basedir=`dirname "$0"`
Index: node-v10.10.0/deps/npm/bin/npx-cli.js
===================================================================
--- node-v10.10.0.orig/deps/npm/bin/npx-cli.js
+++ node-v10.10.0/deps/npm/bin/npx-cli.js
@@ -1,4 +1,4 @@
-#!/usr/bin/env node
+#!/usr/bin/node10
 
 const npx = require('libnpx')
 const path = require('path')
Index: node-v10.10.0/deps/npm/node_modules/mkdirp/bin/cmd.js
===================================================================
--- node-v10.10.0.orig/deps/npm/node_modules/mkdirp/bin/cmd.js
+++ node-v10.10.0/deps/npm/node_modules/mkdirp/bin/cmd.js
@@ -1,5 +1,3 @@
-#!/usr/bin/env node
-
 var mkdirp = require('../');
 var minimist = require('minimist');
 var fs = require('fs');
Index: node-v10.10.0/deps/npm/node_modules/node-gyp/gyp/gyp
===================================================================
--- node-v10.10.0.orig/deps/npm/node_modules/node-gyp/gyp/gyp
+++ node-v10.10.0/deps/npm/node_modules/node-gyp/gyp/gyp
@@ -1,4 +1,3 @@
-#!/bin/sh
 # Copyright 2013 The Chromium Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
Index: node-v10.10.0/deps/npm/node_modules/node-gyp/gyp/gyp_main.py
===================================================================
--- node-v10.10.0.orig/deps/npm/node_modules/node-gyp/gyp/gyp_main.py
+++ node-v10.10.0/deps/npm/node_modules/node-gyp/gyp/gyp_main.py
@@ -1,5 +1,3 @@
-#!/usr/bin/env python
-
 # Copyright (c) 2009 Google Inc. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
Index: node-v10.10.0/deps/npm/node_modules/node-gyp/gyp/pylib/gyp/__init__.py
===================================================================
--- node-v10.10.0.orig/deps/npm/node_modules/node-gyp/gyp/pylib/gyp/__init__.py
+++ node-v10.10.0/deps/npm/node_modules/node-gyp/gyp/pylib/gyp/__init__.py
@@ -1,5 +1,3 @@
-#!/usr/bin/env python
-
 # Copyright (c) 2012 Google Inc. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
Index: node-v10.10.0/deps/npm/node_modules/node-gyp/gyp/pylib/gyp/flock_tool.py
===================================================================
--- node-v10.10.0.orig/deps/npm/node_modules/node-gyp/gyp/pylib/gyp/flock_tool.py
+++ node-v10.10.0/deps/npm/node_modules/node-gyp/gyp/pylib/gyp/flock_tool.py
@@ -1,4 +1,3 @@
-#!/usr/bin/env python
 # Copyright (c) 2011 Google Inc. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
Index: node-v10.10.0/deps/npm/node_modules/node-gyp/gyp/pylib/gyp/mac_tool.py
===================================================================
--- node-v10.10.0.orig/deps/npm/node_modules/node-gyp/gyp/pylib/gyp/mac_tool.py
+++ node-v10.10.0/deps/npm/node_modules/node-gyp/gyp/pylib/gyp/mac_tool.py
@@ -1,4 +1,3 @@
-#!/usr/bin/env python
 # Copyright (c) 2012 Google Inc. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
Index: node-v10.10.0/deps/npm/node_modules/node-gyp/gyp/pylib/gyp/win_tool.py
===================================================================
--- node-v10.10.0.orig/deps/npm/node_modules/node-gyp/gyp/pylib/gyp/win_tool.py
+++ node-v10.10.0/deps/npm/node_modules/node-gyp/gyp/pylib/gyp/win_tool.py
@@ -1,5 +1,3 @@
-#!/usr/bin/env python
-
 # Copyright (c) 2012 Google Inc. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
Index: node-v10.10.0/deps/npm/node_modules/node-gyp/gyp/samples/samples
===================================================================
--- node-v10.10.0.orig/deps/npm/node_modules/node-gyp/gyp/samples/samples
+++ node-v10.10.0/deps/npm/node_modules/node-gyp/gyp/samples/samples
@@ -1,5 +1,3 @@
-#!/usr/bin/python
-
 # Copyright (c) 2009 Google Inc. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
Index: node-v10.10.0/deps/npm/node_modules/node-gyp/gyp/setup.py
===================================================================
--- node-v10.10.0.orig/deps/npm/node_modules/node-gyp/gyp/setup.py
+++ node-v10.10.0/deps/npm/node_modules/node-gyp/gyp/setup.py
@@ -1,5 +1,3 @@
-#!/usr/bin/env python
-
 # Copyright (c) 2009 Google Inc. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
Index: node-v10.10.0/deps/npm/node_modules/node-gyp/gyp/tools/graphviz.py
===================================================================
--- node-v10.10.0.orig/deps/npm/node_modules/node-gyp/gyp/tools/graphviz.py
+++ node-v10.10.0/deps/npm/node_modules/node-gyp/gyp/tools/graphviz.py
@@ -1,5 +1,3 @@
-#!/usr/bin/env python
-
 # Copyright (c) 2011 Google Inc. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
Index: node-v10.10.0/deps/npm/node_modules/node-gyp/gyp/tools/pretty_gyp.py
===================================================================
--- node-v10.10.0.orig/deps/npm/node_modules/node-gyp/gyp/tools/pretty_gyp.py
+++ node-v10.10.0/deps/npm/node_modules/node-gyp/gyp/tools/pretty_gyp.py
@@ -1,5 +1,3 @@
-#!/usr/bin/env python
-
 # Copyright (c) 2012 Google Inc. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
Index: node-v10.10.0/deps/npm/node_modules/node-gyp/gyp/tools/pretty_sln.py
===================================================================
--- node-v10.10.0.orig/deps/npm/node_modules/node-gyp/gyp/tools/pretty_sln.py
+++ node-v10.10.0/deps/npm/node_modules/node-gyp/gyp/tools/pretty_sln.py
@@ -1,5 +1,3 @@
-#!/usr/bin/env python
-
 # Copyright (c) 2012 Google Inc. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
Index: node-v10.10.0/deps/npm/node_modules/node-gyp/gyp/tools/pretty_vcproj.py
===================================================================
--- node-v10.10.0.orig/deps/npm/node_modules/node-gyp/gyp/tools/pretty_vcproj.py
+++ node-v10.10.0/deps/npm/node_modules/node-gyp/gyp/tools/pretty_vcproj.py
@@ -1,5 +1,3 @@
-#!/usr/bin/env python
-
 # Copyright (c) 2012 Google Inc. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
Index: node-v10.10.0/deps/npm/node_modules/node-gyp/node_modules/nopt/bin/nopt.js
===================================================================
--- node-v10.10.0.orig/deps/npm/node_modules/node-gyp/node_modules/nopt/bin/nopt.js
+++ node-v10.10.0/deps/npm/node_modules/node-gyp/node_modules/nopt/bin/nopt.js
@@ -1,4 +1,3 @@
-#!/usr/bin/env node
 var nopt = require("../lib/nopt")
   , path = require("path")
   , types = { num: Number
Index: node-v10.10.0/deps/npm/node_modules/node-gyp/node_modules/semver/bin/semver
===================================================================
--- node-v10.10.0.orig/deps/npm/node_modules/node-gyp/node_modules/semver/bin/semver
+++ node-v10.10.0/deps/npm/node_modules/node-gyp/node_modules/semver/bin/semver
@@ -1,4 +1,3 @@
-#!/usr/bin/env node
 // Standalone semver comparison program.
 // Exits successfully and prints matching version(s) if
 // any supplied version is valid and passes all tests.
Index: node-v10.10.0/deps/npm/node_modules/nopt/bin/nopt.js
===================================================================
--- node-v10.10.0.orig/deps/npm/node_modules/nopt/bin/nopt.js
+++ node-v10.10.0/deps/npm/node_modules/nopt/bin/nopt.js
@@ -1,4 +1,3 @@
-#!/usr/bin/env node
 var nopt = require("../lib/nopt")
   , path = require("path")
   , types = { num: Number
Index: node-v10.10.0/deps/npm/node_modules/qrcode-terminal/bin/qrcode-terminal.js
===================================================================
--- node-v10.10.0.orig/deps/npm/node_modules/qrcode-terminal/bin/qrcode-terminal.js
+++ node-v10.10.0/deps/npm/node_modules/qrcode-terminal/bin/qrcode-terminal.js
@@ -1,5 +1,3 @@
-#!/usr/bin/env node
-
 /*!
  * Module dependencies.
  */
Index: node-v10.10.0/deps/npm/node_modules/rimraf/bin.js
===================================================================
--- node-v10.10.0.orig/deps/npm/node_modules/rimraf/bin.js
+++ node-v10.10.0/deps/npm/node_modules/rimraf/bin.js
@@ -1,5 +1,3 @@
-#!/usr/bin/env node
-
 var rimraf = require('./')
 
 var help = false
Index: node-v10.10.0/deps/npm/node_modules/semver/bin/semver
===================================================================
--- node-v10.10.0.orig/deps/npm/node_modules/semver/bin/semver
+++ node-v10.10.0/deps/npm/node_modules/semver/bin/semver
@@ -1,4 +1,3 @@
-#!/usr/bin/env node
 // Standalone semver comparison program.
 // Exits successfully and prints matching version(s) if
 // any supplied version is valid and passes all tests.
Index: node-v10.10.0/deps/npm/node_modules/uuid/bin/uuid
===================================================================
--- node-v10.10.0.orig/deps/npm/node_modules/uuid/bin/uuid
+++ node-v10.10.0/deps/npm/node_modules/uuid/bin/uuid
@@ -1,4 +1,3 @@
-#!/usr/bin/env node
 var assert = require('assert');
 
 function usage() {
Index: node-v10.10.0/deps/npm/node_modules/which/bin/which
===================================================================
--- node-v10.10.0.orig/deps/npm/node_modules/which/bin/which
+++ node-v10.10.0/deps/npm/node_modules/which/bin/which
@@ -1,4 +1,3 @@
-#!/usr/bin/env node
 var which = require("../")
 if (process.argv.length < 3)
   usage()
Index: node-v10.10.0/deps/npm/bin/node-gyp-bin/node-gyp
===================================================================
--- node-v10.10.0.orig/deps/npm/bin/node-gyp-bin/node-gyp
+++ node-v10.10.0/deps/npm/bin/node-gyp-bin/node-gyp
@@ -1,4 +1,4 @@
-#!/usr/bin/env sh
+#!/bin/sh
 if [ "x$npm_config_node_gyp" = "x" ]; then
   node "`dirname "$0"`/../../node_modules/node-gyp/bin/node-gyp.js" "$@"
 else
Index: node-v10.10.0/deps/npm/node_modules/node-gyp/bin/node-gyp.js
===================================================================
--- node-v10.10.0.orig/deps/npm/node_modules/node-gyp/bin/node-gyp.js
+++ node-v10.10.0/deps/npm/node_modules/node-gyp/bin/node-gyp.js
@@ -1,4 +1,4 @@
-#!/usr/bin/env node
+#!/usr/bin/node10
 
 /**
  * Set the title.
Index: node-v10.10.0/deps/npm/node_modules/npm-lifecycle/node-gyp-bin/node-gyp
===================================================================
--- node-v10.10.0.orig/deps/npm/node_modules/npm-lifecycle/node-gyp-bin/node-gyp
+++ node-v10.10.0/deps/npm/node_modules/npm-lifecycle/node-gyp-bin/node-gyp
@@ -1,4 +1,4 @@
-#!/usr/bin/env sh
+#!/bin/sh
 if [ "x$npm_config_node_gyp" = "x" ]; then
   node "`dirname "$0"`/../../node_modules/node-gyp/bin/node-gyp.js" "$@"
 else
Index: node-v10.10.0/deps/npm/lib/utils/completion.sh
===================================================================
--- node-v10.10.0.orig/deps/npm/lib/utils/completion.sh
+++ node-v10.10.0/deps/npm/lib/utils/completion.sh
@@ -1,4 +1,3 @@
-#!/bin/bash
 ###-begin-npm-completion-###
 #
 # npm command completion script
Index: node-v10.10.0/deps/npm/configure
===================================================================
--- node-v10.10.0.orig/deps/npm/configure
+++ node-v10.10.0/deps/npm/configure
@@ -1,5 +1,3 @@
-#!/bin/bash
-
 # set configurations that will be "sticky" on this system,
 # surviving npm self-updates.
 
Index: node-v10.10.0/deps/npm/node_modules/JSONStream/bin.js
===================================================================
--- node-v10.10.0.orig/deps/npm/node_modules/JSONStream/bin.js
+++ node-v10.10.0/deps/npm/node_modules/JSONStream/bin.js
@@ -1,4 +1,3 @@
-#! /usr/bin/env node
 
 var JSONStream = require('./')
 
Index: node-v10.10.0/deps/npm/node_modules/errno/build.js
===================================================================
--- node-v10.10.0.orig/deps/npm/node_modules/errno/build.js
+++ node-v10.10.0/deps/npm/node_modules/errno/build.js
@@ -1,4 +1,3 @@
-#!/usr/bin/env node
 
 var request   = require('request')
   , fs        = require('fs')
@@ -40,4 +39,4 @@ request(uvheadloc, function (err, respon
   out += '\nmodule.exports.custom = require("./custom")(module.exports)\n'
 
   fs.writeFile('errno.js', out)
-})
\ No newline at end of file
+})
Index: node-v10.10.0/deps/npm/node_modules/errno/cli.js
===================================================================
--- node-v10.10.0.orig/deps/npm/node_modules/errno/cli.js
+++ node-v10.10.0/deps/npm/node_modules/errno/cli.js
@@ -1,4 +1,3 @@
-#!/usr/bin/env node
 
 var errno = require('./')
   , arg   = process.argv[2]
Index: node-v10.10.0/deps/npm/node_modules/is-ci/bin.js
===================================================================
--- node-v10.10.0.orig/deps/npm/node_modules/is-ci/bin.js
+++ node-v10.10.0/deps/npm/node_modules/is-ci/bin.js
@@ -1,4 +1,3 @@
-#!/usr/bin/env node
 'use strict'
 
 process.exit(require('./') ? 0 : 1)
Index: node-v10.10.0/deps/npm/node_modules/rc/cli.js
===================================================================
--- node-v10.10.0.orig/deps/npm/node_modules/rc/cli.js
+++ node-v10.10.0/deps/npm/node_modules/rc/cli.js
@@ -1,4 +1,3 @@
-#! /usr/bin/env node
 var rc = require('./index')
 
 console.log(JSON.stringify(rc(process.argv[2]), false, 2))
Index: node-v10.10.0/deps/npm/node_modules/sshpk/bin/sshpk-conv
===================================================================
--- node-v10.10.0.orig/deps/npm/node_modules/sshpk/bin/sshpk-conv
+++ node-v10.10.0/deps/npm/node_modules/sshpk/bin/sshpk-conv
@@ -1,4 +1,3 @@
-#!/usr/bin/env node
 // -*- mode: js -*-
 // vim: set filetype=javascript :
 // Copyright 2015 Joyent, Inc.	All rights reserved.
Index: node-v10.10.0/deps/npm/node_modules/sshpk/bin/sshpk-sign
===================================================================
--- node-v10.10.0.orig/deps/npm/node_modules/sshpk/bin/sshpk-sign
+++ node-v10.10.0/deps/npm/node_modules/sshpk/bin/sshpk-sign
@@ -1,4 +1,3 @@
-#!/usr/bin/env node
 // -*- mode: js -*-
 // vim: set filetype=javascript :
 // Copyright 2015 Joyent, Inc.  All rights reserved.
Index: node-v10.10.0/deps/npm/node_modules/sshpk/bin/sshpk-verify
===================================================================
--- node-v10.10.0.orig/deps/npm/node_modules/sshpk/bin/sshpk-verify
+++ node-v10.10.0/deps/npm/node_modules/sshpk/bin/sshpk-verify
@@ -1,4 +1,3 @@
-#!/usr/bin/env node
 // -*- mode: js -*-
 // vim: set filetype=javascript :
 // Copyright 2015 Joyent, Inc.  All rights reserved.
